frontend/src/shared/video-card.tsx

```diff
@@ -0,0 +1,34 @@
+"use client";
+
+import clsx from "clsx";
+import { Video } from "@/shared/api";
+
+interface Props {
+  video: Video;
+}
+
+export default function VideoCard({ video }: Props) {
+  const figureClassname = clsx({
+    "flex-auto grow self-center pt-8 glass w-full h-full": video.processing,
+  });
+
+  const cardBodyClassname = clsx({
+    "card-body justify-end": video.processing,
+    "card-body": !video.processing,
+  });
+
+  return (
+    <div className="card card-compact w-96 h-full bg-base-100 shadow-xl min-w-[28rem] min-h-[20rem]">
+      <figure className={figureClassname}>
+        {video.processing ? <div>Processing!</div> : <img src={`/api/clips/${video.id}/thumbnail.jpg`} />}
+      </figure>
+      <div className={cardBodyClassname}>
+        <div className="flex-row flex text-xl">
+          <div className="container flex card-title">{video.title}</div>
+          <div className="container flex card-title justify-end">{video.views}</div>
```

Comment:
Should we do a youtube style `123 views` or maybe do an eyeball icon near the number? I feel like it's not quite obvious enough when you're looking at a video card what the random `3` in the corner means.

Reply:
👍 That is a great idea, I was going to add it in a subsequent PR or delegate the work to @CameronLehr since it would be a good intro into the `Intl` API provided by browsers. I'll add it in this PR though since it'll be a large improvement

Reply:
![image](https://user-images.githubusercontent.com/10362744/218365033-67974d5c-ba88-431a-943d-cfe262b26c1f.png)

I've added a nicer view count here and also the upload date of the video

I eventually want to add something like `X days ago` or `Y minutes ago` but that would require a bit more thought on how to handle that properly and is probably outside of the scope of this PR